src/sources.cpp

```diff
@@ -299,26 +274,22 @@ static void src_vol_chunkloop(fields_chunk *fc, int ichunk, component c, ivec is
 
   if (idx_vol > npts) meep::abort("add_volume_source: computed wrong npts (%zd vs. %zd)", npts, idx_vol);
 
-  src_vol *tmp = new src_vol(c, data->src, idx_vol, index_array, amps_array);
   field_type ft = is_magnetic(c) ? B_stuff : D_stuff;
-  fc->sources[ft] = tmp->add_to(fc->sources[ft]);
+  fc->add_source(ft, src_vol(c, data->src, std::move(index_array), std::move(amps_array)));
```

Comment:
Isn't `std::move` already called in the `src_vol` constructor?

Reply:
It is needed in both places to turn lvalues into xvalues.
If `std::move` were removed here, compilation would fail because a `vector<...>` cannot be converted to the rvalue reference `vector<...> &&`.
In the `src_vol` constructor, `std::move` is used to select the correct overload of the the `vector` constructor. If removed, compilation would succeed but the vector data would be copied instead of moved.

See the "Notes" section here: https://en.cppreference.com/w/cpp/utility/move